ladybug_geometry/geometry3d/cone.py

```diff
@@ -0,0 +1,191 @@
+
+        Args:
+            axis: A Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in radians.
+            origin: A Point3D for the origin around which the sphere will be rotated.
+        """
+        return Cone3D(self.vertex.rotate(axis, angle, origin),
+                      self.axis.rotate(axis, angle, origin),
+                      self.angle,
+                      self.height)
+
+    def rotate_xy(self, angle, origin):
+        """Get a cone that is rotated counterclockwise in the world XY plane by an angle.
+
+        Args:
+            angle: An angle for rotation in radians.
+            origin: A Point3D for the origin around which the sphere will be rotated.
+        """
+        return Cone3D(self.vertex.rotate_xy(angle, origin),
+                      self.axis.rotate_xy(angle, origin),
+                      self.angle,
+                      self.height)
+
+    def reflect(self, normal, origin):
+        """Get a cone reflected across a plane with the input normal vector and origin.
+
+        Args:
+            normal: A Vector3D representing the normal vector for the plane across
+                which the arc will be reflected. THIS VECTOR MUST BE NORMALIZED.
+            origin: A Point3D representing the origin from which to reflect.
+        """
+        return Cone3D(self.vertex.reflect(normal, origin),
+                      self.axis.reflect(normal, origin),
+                      self.angle,
+                      self.height)
+
+    def scale(self, factor, origin=None):
+        """Scale a cone by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the cone should be scaled.
+            origin: A Point3D representing the origin from which to scale.
+                If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        return Cone3D(self.origin.scale(factor, origin),
+                      self.axis,
+                      self.angle,
+                      self.height)
+
+    def duplicate(self):
+        """Get a copy of this object."""
+        return self.__copy__()
+
+    def to_dict(self):
+        """Get Cone as a dictionary."""
+        return {'type': 'Cone3D', 'vertex': self.vertex.to_array(),
+                'axis': self.axis.to_array(), 'angle': self.angle, 'height': self.height}
+
+    def __copy__(self):
+        return Cone3D(self.vertex, self.axis, self.angle, self.height)
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        return 'Cone3D (vertex {}) (axis {}) (angle {}) (height {}))'.format(self.vertex,
+                                                                             self.axis,
+                                                                             self.angle,
+                                                                             self.height)
```

Comment:
I'd recommend putting all of the properties inside the `format` bracket on one line. As long as you indent it from the one above it, this should work fine and be a bit more readable.

Reply:
Yep, my bad!